app/models/repository.rb

```diff
@@ -249,6 +250,19 @@ def unassign_unshared_items
                          .destroy_all
   end
 
+  def unlink_unshared_items
+    RepositoryRowConnection.transaction do
+      repository_rows_to_unlink = RepositoryRow.joins(:repository)
```

Comment:
So basically here all other rows get selected? It's better to limit the scope to only needed rows